examples/3d-tiles/tileset-3d-layer.js

```diff
@@ -203,43 +249,55 @@ export default class Tileset3DLayer extends CompositeLayer {
           normals: {value: positions, size: 3},
           length: positions.length / 3
         },
-        ...coordinateProps,
         numInstances: pointsCount,
-        getPosition: (object, {index, data, target}) => {
-          target[0] = data.positions[index * 3];
-          target[1] = data.positions[index * 3 + 1];
-          target[2] = data.positions[index * 3 + 2];
-
-          if (transform && coordinateProps.coordinateSystem === COORDINATE_SYSTEM.LNGLAT) {
-            target = new Matrix4().multiplyRight(transform).transformVector3(target);
-            Ellipsoid.WGS84.cartesianToCartographic(target, target);
-          }
-
-          return target;
-        },
-        getColor: (...args) => {
-          return this._getColor(tileHeader, ...args);
-        },
-        getNormal: normals
-          ? (object, {index, data, target}) => {
-              target[0] = data.normals[index * 3];
-              target[1] = data.normals[index * 3 + 1];
-              target[2] = data.normals[index * 3 + 2];
-              return target;
+        ...positionProps,
+        getColor: colors
+          ? (...args) => {
+              return this._getColor(tileHeader, ...args);
             }
-          : [0, 1, 0],
+          : color || [255, 255, 0, 200],
+        getNormal: normals ? (...args) => this._getNormal(...args) : [0, 1, 0],
         opacity: 0.8,
-        pointSize: 1.5
+        pointSize: 1.5,
+        ...transformProps
       })
     );
   }
 
-  /* eslint-disable max-statements */
-  _getColor(tileHeader, object, {index, data, target}) {
-    if (!tileHeader) {
-      return null;
+  _getPosition(tileHeader, object, {index, data, target}) {
+    const {transform} = tileHeader.userData;
+    const {rtcCenter} = tileHeader.content;
+    const {isWGS84} = this.props;
+    target[0] = data.positions[index * 3];
+    target[1] = data.positions[index * 3 + 1];
+    target[2] = data.positions[index * 3 + 2];
+
+    // TODO
+    // How to tell if this is WGS84 crs or not?
+    // How to transform data point using GPU?
+    if (isWGS84) {
+      let matrix = new Matrix4();
```

Comment:
This function is getting called once for every position. Creating the matrix in here will surely kill performance. You already create matrices outside...